oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsGlobalMemory.java

```diff
@@ -1,102 +1,194 @@
+
+import com.sun.jna.platform.win32.Kernel32; // NOSONAR squid:S1191
+import com.sun.jna.platform.win32.Psapi;
+import com.sun.jna.platform.win32.Psapi.PERFORMANCE_INFORMATION;
+import com.sun.jna.platform.win32.COM.WbemcliUtil.WmiQuery;
+import com.sun.jna.platform.win32.COM.WbemcliUtil.WmiResult;
+
+import oshi.hardware.PhysicalMemory;
+import oshi.hardware.VirtualMemory;
+import oshi.hardware.common.AbstractGlobalMemory;
+import oshi.util.platform.windows.WmiQueryHandler;
+import oshi.util.platform.windows.WmiUtil;
+
+/**
+ * Memory obtained by Performance Info.
+ */
+public class WindowsGlobalMemory extends AbstractGlobalMemory {
+
+    private static final Logger LOG = LoggerFactory.getLogger(WindowsGlobalMemory.class);
+
+    private final Supplier<PerfInfo> perfInfo = memoize(this::readPerfInfo, defaultExpiration());
+
+    private final Supplier<VirtualMemory> vm = memoize(this::createVirtualMemory);
+    
+    enum PhysicalMemoryProperty{
+	BANKLABEL,CAPACITY,CONFIGUREDCLOCKSPEED,MANUFACTURER,MEMORYTYPE
+    }
+
+    @Override
+    public long getAvailable() {
+        return perfInfo.get().available;
+    }
+
+    @Override
+    public long getTotal() {
+        return perfInfo.get().total;
+    }
+
+    @Override
+    public long getPageSize() {
+        return perfInfo.get().pageSize;
+    }
+
+    @Override
+    public VirtualMemory getVirtualMemory() {
+        return vm.get();
+    }
+
+    private VirtualMemory createVirtualMemory() {
+	
+        return new WindowsVirtualMemory(getPageSize());
+    }
+    
+    @Override
+    public List<PhysicalMemory> getPhysicalMemory() {
+	WmiQuery<PhysicalMemoryProperty> physicalMemoryQuery = new WmiQuery<>("Win32_PhysicalMemory", PhysicalMemoryProperty.class);
+	WmiQueryHandler wmiQueryHandler = WmiQueryHandler.createInstance();
+	WmiResult<PhysicalMemoryProperty> bankMap = wmiQueryHandler.queryWMI(physicalMemoryQuery);
+	List<PhysicalMemory> physicalMemoryArray =new ArrayList<>();
+	PhysicalMemory memory = null;
+	if (bankMap.getResultCount() > 0) {
+	    //loop over the results if memory consists of more than one bank
+	    for(int index =0;index < bankMap.getResultCount();index++) {
+		String bankLabel = WmiUtil.getString(bankMap, PhysicalMemoryProperty.BANKLABEL, index);
+		long capacity = WmiUtil.getUint64(bankMap, PhysicalMemoryProperty.CAPACITY, index);
+		long speed = WmiUtil.getUint32(bankMap, PhysicalMemoryProperty.CONFIGUREDCLOCKSPEED, index);
+		String manufacturer = WmiUtil.getString(bankMap, PhysicalMemoryProperty.MANUFACTURER, index);
+		int type = WmiUtil.getUint16(bankMap, PhysicalMemoryProperty.MEMORYTYPE, index);
+		String memoryType = convertMemoryTypeToString(type);
+		memory = new PhysicalMemory(bankLabel, capacity, speed, manufacturer, memoryType);
+		physicalMemoryArray.add(memory);
+    	   }
+	}
+   	return physicalMemoryArray;
+   }
+    
+    public static String convertMemoryTypeToString(int type) {
+	switch(type) {
+        	case 0:

```

Comment:
Interestingly, this seems to be "0" for most modern systems, and I'm wondering if this WMI field is even useful to retrieve any more.

Reply:
OK, so I finally had the time to do some testing on my physical Win10 machine.  The `MemoryType` field has a limited enum set and is useless on modern hardware (my system returns 0).  I just committed a change that uses the `SMBIOSMemoryType` field instead, which is 26 (0x1A) on my system.  [This StackOverflow answer](https://stackoverflow.com/a/14227777/1161484) indicates this is DDR4.  I haven't managed to find the source document for this, but it's a better mapping to use here.

I made a few formatting tweaks with my commit as well.

Reply:
Also it appears that SMBIOS type 17 mapping is cross platform so you might want to toss this method into the abstract class (and put it back to `protected`, sorry!)

Reply:
[Source here](https://www.dmtf.org/sites/default/files/standards/documents/DSP0134_3.2.0.pdf), page 98